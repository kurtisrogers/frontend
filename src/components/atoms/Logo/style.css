.logo {
  position: relative;
  text-transform: uppercase;
  text-decoration: none;
  text-rendering: optimizeLegibility;
  outline: 2px solid transparent;
  padding: 1rem 0.25rem 0px 2rem;
  -webkit-user-select: none; /* Safari */
  user-select: none;
  text-align: end;
}

.logo > small {
  display: block;
}

.logo:before {
  content: "";
  position: absolute;
  left: 2px;
  top: 2px;
  height: 5px;
  width: 5px;
  border-top: 2px solid;
  border-left: 2px solid;
}
