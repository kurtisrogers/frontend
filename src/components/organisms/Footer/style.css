footer {
  background: var(--brand-yellow);
  background: var(--brand-primary-gradient);
  padding: 4rem 1rem;
  max-width: var(--brand-base-grid);
  margin: 1rem;
}

footer .filterPanel--blur {
  display: flex;
  justify-content: space-between;
}

footer .logo {
  display: block;
  margin-bottom: 0.625rem;
}

footer nav:not(.socials) {
  text-align: right;
}

footer nav ul {
  display: block;
  margin: 0px;
}

footer nav:not(.socials) li {
  list-style-type: none;
  margin-bottom: 0.5rem;
}

footer nav a {
  display: inline-flex;
  color: var(--brand-white);
  padding: 0px;
  outline: none;
  font-weight: 600;
  text-underline-offset: 0.125rem;
  text-underline-position: under;
}

footer div.filterPanel > :first-child {
  width: calc(100% - 4rem);
}

footer .content .footer__socials {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

footer .socials ul {
  display: flex;
  flex-direction: row;
  margin: 0px;
}

@media screen and (min-width: 1280px) {
  footer {
    margin: 0px auto 1rem;
  }
}
