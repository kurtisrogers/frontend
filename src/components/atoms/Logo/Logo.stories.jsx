import Logo from "./index";
import "./style.css";

export default {
  title: "Atoms/Logo",
  component: Logo,
  argTypes: {
    variant: {
      options: ['primary', 'secondary'],
      control: { type: "radio" }
    },
    text: {
      control: { type: "text" }
    },
    isButton: {
      control: { type: "none" }
    }
  },
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen"
  },
};

export const Default = {
  args: {
    isButton: true,
    variant: "primary",
    text: "Kurtis Rogers",
  },
  render: (args) => <Logo {...args} />
};
