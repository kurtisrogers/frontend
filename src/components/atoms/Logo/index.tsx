import { A } from "@solidjs/router";
import "./style.css";

interface Props {
  variant?: "primary" | "secondary";
  text?: string;
}

export default function Logo({ variant = "primary", text = "Kurtis Rogers" }: Props) {
  const variantStyle = variant === "primary" ? "bg-red text-white" : "bg-white text-black";

  const windowWidth = window.innerWidth ?? 769;
  const handleLogoStringResize =
    windowWidth > 768
      ? text
      : text
          .split("")
          .filter(a => a.match(/[A-Z]/))
          .join("");

  return (
    <A href="/" class={`logo ${variantStyle}`} aria-label="Go to the homepage">
      <strong>{handleLogoStringResize}</strong>
    </A>
  );
}
