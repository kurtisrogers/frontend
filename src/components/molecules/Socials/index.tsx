import Icon from "@/components/atoms/Icon";
import "./style.css";
import icons from "@/data/icons";

export default function Socials() {
  return (
    <nav class="socials">
      <ul>
        <li>
          <a href="https://www.linkedin.com/in/kurtisrogers/" target="_blank">
            <Icon width={50} height={50} viewbox="0 0 24 24" icondata={icons.linkedin} />
            <span class="sr-only">A link to my LinkedIn profile (opens a new tab)</span>
          </a>
        </li>
        <li>
          <a href="https://github.com/Kurtmcmurt/kurtisrogers.com/tree/develop" target="_blank">
            <Icon width={50} height={50} viewbox="0 0 24 24" icondata={icons.github} />
            <span class="sr-only">A link to the repo for this website (opens a new tab)</span>
          </a>
        </li>
      </ul>
    </nav>
  );
}
