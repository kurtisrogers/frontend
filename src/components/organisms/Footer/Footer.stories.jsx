import Footer from "./index";
import "./style.css";

export default {
  title: "Organisms/Footer",
  component: Footer,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/solid/writing-docs/docs-page
  tags: ["autodocs"],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/solid/configure/story-layout
    layout: "fullscreen"
  }
};

const Template = () => <Footer />;

export const Default = Template.bind({});
// Default.args = {
//   // Add any props you want to pass to the Header component
// };
