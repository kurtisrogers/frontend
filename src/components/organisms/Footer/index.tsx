import { JSX } from "solid-js";
import "./style.css";

interface Props {
  children?: JSX.Element;
}

export default function Footer({ children }: Readonly<Props>) {
  return (
    <footer tabindex="-1">
      <div class="content">
        {/* some other stuff here? */}
        {children}
      </div>
    </footer>
  );
}
