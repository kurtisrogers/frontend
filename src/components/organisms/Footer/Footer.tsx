import "./Footer.css";

export default function Footer(props: any) {
  return (
    <footer>
      <p>Hello I'm the footer</p>
      {props.children}
    </footer>
  )
}
