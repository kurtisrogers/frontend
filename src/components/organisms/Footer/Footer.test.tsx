import { render } from "@solidjs/testing-library";
import Footer from ".";
import Socials from "../../molecules/Socials";
import Logo from "../../atoms/Logo";
import { describe, it, expect } from "vitest";

vi.mock("@solidjs/router", async importOriginal => {
  const actual = (await importOriginal()) ?? {};
  return {
    ...actual,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    A: (props: any) => {
      const { children, ...rest } = props;
      return (
        <a {...rest} data-testid="mock-link">
          {children}
        </a>
      );
    }
  };
});

describe("<Footer />", () => {
  const renderFooter = () =>
    render(() => (
      <Footer>
        <Logo variant="secondary" />
        <Socials />
      </Footer>
    ));

  it("shows a logo for the website", () => {
    const { getByText } = renderFooter();
    const logo = getByText(/Kurtis Rogers/);

    expect(logo).toBeInTheDocument();
  });

  it("has a link to my linkedin profile", () => {
    const { getByText } = renderFooter();
    const linkedinLink = getByText("A link to my LinkedIn profile (opens a new tab)");

    expect(linkedinLink).toMatchSnapshot();
    expect(linkedinLink).toBeInTheDocument();
  });

  it("has a link to the repo for the website", () => {
    const { getByText } = renderFooter();
    const githubLink = getByText("A link to the repo for this website (opens a new tab)");

    expect(githubLink).toMatchSnapshot();
    expect(githubLink).toBeInTheDocument();
  });
});
