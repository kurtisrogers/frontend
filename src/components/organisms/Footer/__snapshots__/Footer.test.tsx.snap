// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Footer /> > has a link to my linkedin profile 1`] = `
<span
  class="sr-only"
>
  A link to my LinkedIn profile (opens a new tab)
</span>
`;

exports[`<Footer /> > has a link to the repo for the website 1`] = `
<span
  class="sr-only"
>
  A link to the repo for this website (opens a new tab)
</span>
`;
